webui/react/src/components/ExperimentActions.tsx

```diff
@@ -0,0 +1,169 @@
+import { Button, Popconfirm } from 'antd';
+import React, { useCallback, useState } from 'react';
+
+import { archiveExperiment, killExperiment, launchTensorboard, setExperimentState,
+} from 'services/api';
+import { ExperimentDetails, RunState, TBSourceType } from 'types';
+import { openCommand } from 'utils/routes';
+import { cancellableRunStates, killableRunStates, terminalRunStates } from 'utils/types';
+
+import css from './ExperimentActions.module.scss';
+
+interface Props {
+  experiment: ExperimentDetails;
+  finally: () => void; // A callback to trigger after an action is done.
+}
+
+const ExperimentActions: React.FC<Props> = ({ experiment, finally: updateFn }: Props) => {
+
+  interface ButtonLoadingStates {
+    kill: boolean;
+    [RunState.StoppingCanceled]: boolean;
+    archive: boolean;
+    [RunState.Paused]: boolean;
+    [RunState.Active]: boolean;
+    tsb: boolean; // tensorboard
```

Comment:
**suggestion:** Create a separate enum just for experiment actions. In this [PR](https://github.com/determined-ai/determined/pull/866/files) I defined something like this:
```
enum Action {
  Activate = 'Activate',
  Archive = 'Archive',
  Cancel = 'Cancel',
  Kill = 'Kill',
  Pause = 'Pause',
  OpenTensorBoard = 'Open Tensorboard',
}
```

Reply:
this is a lot better :+1:

Reply:
how does this look? 

```
enum Action {
  Archive = 'Archive',
  Kill = 'Kill',
  Tensorboard = 'Tensorboard',
}

type ButtonLoadingStates = Record<Action | RunState, boolean>;
```